src/BlockNLPModels.jl

```diff
@@ -0,0 +1,17 @@
+module BlockNLPModels
+
+using NLPModels
+using NLPModelsJuMP
+using Ipopt
+using SolverCore
+using NLPModelsIpopt
+
+
+include("blk_NLP.jl")
```

Comment:
I would advice to use explicit naming, to help the other contributors (explicit is always better than implicit). 
`blk` -> `block`
`dd` -> ??
`Block_Solver.jl` -> `block_solver.jl` (I advise avoid caps when naming files)
`Full_Space` -> `full_space.jl`

Reply:
I will keep this in mind. There are some places where I've already made changes according to this suggestion, but there are others where I don't want to update the code now, because that can break other functions. But I will make sure that everything follow these rules for the 2nd PR.